xcode/Subconscious/Shared/Components/AppView.swift

```diff
@@ -484,6 +495,24 @@ struct AppModel: ModelProtocol {
                 state: state,
                 environment: environment
             )
+        case let .setFirstRunPath(path):
+            var model = state
+            model.firstRunPath = path
+            
+            if path.last == .nickname  {
```

Comment:
Generating this action here seems too implicit/indirect to me.

It might be easier to understand the logical flow if the effect were generated by an action somehow. For example, we could send an action when the nickname "next" navigation link is tapped by decorating the navigation link with `.simultaneousGesture`:

```swift
.simultaneousGesture(TapGesture().onEnded({
    store.send(.createSphere)
}))
```

Reply:
As discussed on call, I moved this into `.onAppear {}` on `FirstStepView` which seems much easier all around.